playbooks/deployments.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Deployments
+description: How systems are deployed at Artsy
+---
+
+# Deployment at Artsy
+
+## Goals
+
+- _Safe_: Only successfully-tested code should be deployed. Deploying should be _graceful_. I.e., shouldn't impact
+  availability of the service.
+- _Repeatable_: The result of a deploy should be consistent no matter the state of the developer's local
+  environment, the particulars of a code change, the state of data, volume of traffic, etc.
+- _Visible_: It should be transparent what deploy is happening when (and by whom...). An audit trail and
+  instrumentation can be helpful in diagnosing production issues and interpreting monitoring results.
+- _Simple_: Other things being equal, we prefer simple deployment operations whose impact can be easily understood
+  by developers who may have to consider how to undo a change or its impact on other systems.
+- _Reversible\*_: We should be able to roll back a deploy while keeping the state of the system stable and
+  consistent. (This may also include associated changes such as to dependencies or configuration.)
+- _Consistent\*_: Ideally the deployment operation is the same whether a change includes just code, updated
+  configuration, data migrations, or dependencies.
+- _Fast\*_: Other things being equal, we prefer deploys to be fast so we can deploy often and focus on other
```

Comment:
Do you mean fast as in "it should not take long for the deploy script to finish"?

One thing that seems to be missing from this list is high frequency-- we should encourage changes to be deployed to production as soon as they can be (early and often?).

Reply:
Good point. It's more about "fresh" and "granular" than "frequent" though. (A stable system may only be deployed infrequently, but should still be deployable shortly after any change.) I tweaked the description a bit.